File: trunk/ivef-objc/Classes/IVEFServerStatus.m

#import "IVEFServerStatus.h"


@implementation IVEFServerStatus

- (id) init {
    self = [super init];
    if (self != nil) {
        m_detailsPresent = false;
        m_dateFormatter = [[NSDateFormatter alloc] init];
        [m_dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS"];
    }
    return self;
}

-(void) setStatus:(NSString *) val {

    if ( ( ![val isEqualToString: @"queuefull"] ) &&
         ( ![val isEqualToString: @"ok"] ) )
        return;
    [m_status release];
    m_status = val;
    [m_status retain];
}

- (NSString *) getStatus {

    return m_status;
}

-(void) setDetails:(NSString *) val {

    m_detailsPresent = true;
    [m_details release];
    m_details = val;
    [m_details retain];
}

- (NSString *) getDetails {

    return m_details;
}

-(bool) hasDetails {

    return m_detailsPresent;
}

-(void) setAttributes:(NSDictionary *)attributeDict {

        for (NSString *key in attributeDict) {
            if ([key isEqualToString: @"Status"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setStatus: val];
            }
            else if ([key isEqualToString:@"Details"]) {
                NSString *val = [attributeDict objectForKey: key];
                [self setDetails: val];
            }
        }
}

-(NSString *) XML {

    NSMutableString *xml = [NSMutableString stringWithString:@"<ServerStatus"];
    [xml appendString: @" Status=\""];
    [xml appendString: m_status];
    [xml appendString: @"\""];
    if ( [self hasDetails] ) {
        [xml appendString: @" Details=\""];
        [xml appendString: m_details];
        [xml appendString: @"\""];
    }
    [xml appendString:@">\n"];
    [xml appendString: @"</ServerStatus>\n"];
    return xml;
}

-(NSString *) stringValueWithLead: (NSString *) lead {

    NSMutableString *str = [[[NSMutableString alloc] init] autorelease];
    [str setString: [lead stringByAppendingString:@"ServerStatus\n"]];
    [str appendString: [lead stringByAppendingString: @" "]];
    [str appendString: @"Status=\""];
    [str appendString: m_status];
    [str appendString: @"\"\n"];

    if ( [self hasDetails] ) {
        [str appendString: [lead stringByAppendingString: @" "]];
        [str appendString: @"Details = \""];
        [str appendString: m_details];
    }
    return str;
}


@end

File: ivef-lib/tags/IVEF_0_1_3/ivef-qt/IVEFServerStatus.cpp

#include "IVEFServerStatus.h"

ServerStatus::ServerStatus() {

    m_status = "";
    m_details = "";
    m_detailsPresent = false;
}

ServerStatus::ServerStatus(const ServerStatus &val) : QObject() {

    m_status = val.m_status;
    m_detailsPresent = val.m_detailsPresent;
    m_details = val.m_details;
}

ServerStatus & ServerStatus::operator=(const ServerStatus &val) {

    m_status = val.m_status;
    m_detailsPresent = val.m_detailsPresent;
    m_details = val.m_details;
    return *this;
}

QString ServerStatus::encode( QString str) {

    str.replace('&', "&amp;");
    str.replace('<', "&lt;");
    str.replace('>', "&gt;");
    str.replace('"', "&quot;");
    return str;
}

void ServerStatus::setStatus(QString val) {

    if ( ( val != "queuefull" ) &&
         ( val != "ok" ) )
        return;
    m_status = val;
}

QString ServerStatus::getStatus() const {

    return m_status;
}

void ServerStatus::setDetails(QString val) {

    m_detailsPresent = true;
    m_details = val;
}

QString ServerStatus::getDetails() const {

    return m_details;
}

bool ServerStatus::hasDetails() {

    return m_detailsPresent;
}

QString ServerStatus::toXML() {

    QString xml = "<ServerStatus";
    xml.append(" Status=\"" + encode (m_status) + "\"");
    if ( hasDetails() ) {
        xml.append(" Details=\"" + encode (m_details) + "\"");
    }
    xml.append(">\n");
    xml.append( "</ServerStatus>\n");
    return xml;
}

QString ServerStatus::toString(QString lead) {

    QString str = lead + "ServerStatus\n";
    str.append( lead + "    Status = " + m_status + "\n");
    if ( hasDetails() ) {
        str.append( lead + "    Details = " + m_details + "\n");
    }
    return str;
}

